internal/tools/docthebuilder/rule/rule_g001.go

```diff
@@ -0,0 +1,38 @@
+package rule
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tools/docthebuilder/data"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tools/docthebuilder/util"
+	"github.com/spf13/afero"
+)
+
+type G001 struct{}
```

Comment:
And then have the type be more indicative of what it is checkingf?
```suggestion
type G001_DocumentationExists struct{}
```
and then the file name can be rule_g001_documentation_exists.go

so that we don't need to memorize rule numbers to know what a type/name/file is